base/libgit2/types.jl

```diff
@@ -52,10 +52,29 @@ end
 
 "Abstract payload type for callback functions"
 abstract AbstractPayload
-user(p::AbstractPayload) = throw(AssertionError("Function 'user' is not implemented for type $(typeof(p))"))
-password(p::AbstractPayload) = throw(AssertionError("Function 'password' is not implemented for type $(typeof(p))"))
-isused(p::AbstractPayload) = throw(AssertionError("Function 'isused' is not implemented for type $(typeof(p))"))
-setused!(p::AbstractPayload,v::Bool) = throw(AssertionError("Function 'setused!' is not implemented for type $(typeof(p))"))
+
+"Abstract credentials payload"
+abstract AbstractCredentials <: AbstractPayload
+"Returns a credentials parameter"
+function Base.getindex(p::AbstractCredentials, keys...)
+    for k in keys
+        ks = Symbol(k)
+        isdefined(p, ks) && return getfield(p, ks)
+    end
+    return nothing
```

Comment:
I did not want to impose any particular structure on `AbstractCredentials` subtypes.